ci/Jenkinsfile.nightly_desktop

```diff
@@ -0,0 +1,306 @@
+external_modules_dir = [
+  'node_modules/react-native-i18n/desktop',
+  'node_modules/react-native-config/desktop',
+  'node_modules/react-native-fs/desktop',
+  'node_modules/react-native-http-bridge/desktop',
+  'node_modules/react-native-webview-bridge/desktop',
+  'node_modules/react-native-keychain/desktop',
+  'node_modules/react-native-securerandom/desktop',
+  'modules/react-native-status/desktop',
+]
+
+def installJSDeps() {
+  def attempt = 1
+  def maxAttempts = 10
+  def installed = false
+  sh 'node -v'
+  sh 'npm -v'
+  while (!installed && attempt <= maxAttempts) {
+    println "#${attempt} attempt to install npm deps"
+    sh 'scripts/prepare-for-platform.sh desktop'
+    sh 'npm install'
+    installed = fileExists('node_modules/web3/index.js')
+    attemp = attempt + 1
+  }
+}
+
+def doGitRebase() {
+  try {
+    sh 'git rebase origin/develop'
+  } catch (e) {
+    sh 'git rebase --abort'
+    throw e
+  }
+}
+
+def cleanupBuild() {
+  sh 'rm -rf node_modules'
+  sh "rm -rf ${packageFolder}"
+  sh 'rm -rf desktop/modules'
+  sh 'rm -rf desktop/node_modules'
+}
+
+def cleanupAndDeps() {
+  cleanupBuild()
+  sh 'cp .env.jenkins .env'
+  sh 'lein deps'
+  installJSDeps()
+}
+
+def slackNotify(message, color = 'good') {
+  slackSend(
+    color: color,
+    channel: '#jenkins-desktop',
+    message: "develop (${env.CHANGE_BRANCH}) ${message} ${env.BUILD_URL}"
+  )
+}
+
+def buildClojureScript() {
+  sh 'rm -f index.desktop.js'
+  sh 'lein prod-build-desktop'
+  sh "mkdir ${packageFolder}"
+  sh """
+    react-native bundle \\
+      --entry-file index.desktop.js \\
+      --dev false --platform desktop \\
+      --bundle-output ${packageFolder}/StatusIm.jsbundle \\
+      --assets-dest ${packageFolder}/assets
+  """
+}
+
+def uploadArtifact(filename) {
+  def domain = 'ams3.digitaloceanspaces.com'
+  def bucket = 'status-im-desktop'
+  withCredentials([usernamePassword(
+    credentialsId: 'digital-ocean-access-keys',
+    usernameVariable: 'DO_ACCESS_KEY',
+    passwordVariable: 'DO_SECRET_KEY'
+  )]) {
+    sh """
+      s3cmd \\
+        --acl-public \\
+        --host='${domain}' \\
+        --host-bucket='%(bucket)s.${domain}' \\
+        --access_key=${DO_ACCESS_KEY} \\
+        --secret_key=${DO_SECRET_KEY} \\
+        put ${filename} s3://${bucket}/
+    """
+
+  }
+  def url = "https://${bucket}.${domain}/${filename}"
+  return url
+}
+
+parallel(
+  'MacOS build': {
+    timeout(90) {
+      node('macos') {
+        load "$HOME/env.groovy"
+
+        try {
+          stage('Git & Deps') {
+            slackNotify('MacOS build started.')
+            checkout scm
+            doGitRebase()
+            cleanupAndDeps()
+            commit = sh(
+              returnStdout: true,
+              script: 'git rev-parse HEAD'
+            ).trim().take(6)
+          }
+
+          stage('Build ClojureScript') {
+            buildClojureScript()
+          }
+
+          stage('Build MacOS binaries') {
```

Comment:
@MaxRis, @jakubgs, here we use `external_modules_dir` which is a copy from package.json.
When we add more records in package.json, for example for external fonts, we will also need to duplicate them here. The same will happen on every change in used modules or fonts.
I think that after committing this version of nightly, it would be nice to try `react-native build-desktop` instead of manual `cmake` invokation. Not sure if it will work right now, probably option for release build isn't supported yet.

Reply:
@vkjr `build-desktop` was not working for me. Probably, native modules were not propagated correctly for some reasons.
Also at the moment we use origin react-native-cli, I believe. It might make sense to update package.json to use our fork of react-native in similar way how it's used when we do it via `init` cli command. I think I have created already some issues for that.

Reply:
https://github.com/status-im/status-react/issues/5405

Reply:
@maxris, thanks.
Btw, I'm not sure how that works, but I have original `react-native-cli` installed globally but when I create new app with it and specify `--version status-im/react-native-desktop`, I'm able to enter in the created directory and invoke `react-native build-desktop`. I believe that local `react-native-cli` somehow used when you are inside the project.

Reply:
@vkjr , yes, I meant to take `package.json` file specifics generated by `--version` and apply these specifics to `package.json` of Desktop project. Hopefully, it will do the trick